common/ts/publish-task.ts

```diff
@@ -28,13 +31,30 @@ export default async function publishTask(endpointType: EndpointType) {
     taskReports.map(taskReport => getReportForTask(taskReport, metrics, endpoint, timeoutSec))
   );
 
+  if (globalQualityGateStatus === '') {
+    globalQualityGateStatus = 'ok';
+  }
+
+  fillBuildProperty();
```

Comment:
It would be worth adding some user-visible logging so users can see what is happening and to help us diagnose issues e.g. `Setting quality gate status for the build to xxx`